### PR TITLE
feat: support reading `cms/sanity.config.ts`

### PR DESCRIPTION
**Pull request by danielroe**
This does several things:

1. by default it looks for a `~~/cms/sanity.config.*` file relative to your project's root directory.
2. it supports explicitly configuring this path with the `configFile` property
3. if the path is within the nuxt project nuxt will automatically reload if the file is edited

It does fall back to the previous behaviour (looking for a `sanity.json` in your root directory) if there is no such file.

### CODE REVIEW

File: src/module.ts [hunk 1]
```diff
@@ -129,16 +128,41 @@ export default defineNuxtModule<SanityModuleOptions>({
...
+  },
   async setup (options, nuxt) {
+    // If explicit configuration is not provided, attempt to load it from `sanity.config.ts`
+    if (!options.projectId || !options.dataset) {
+      // Register watcher on sanity.config.ts
+      const sanityConfigPath = await resolvePath(options.configFile!) || /* backwards compatibility */ resolve(nuxt.options.rootDir, './sanity.json')
+      const relativeSanityConfigPath = relative(nuxt.options.rootDir, sanityConfigPath)
+      if (!relativeSanityConfigPath.startsWith('..')) {
+        nuxt.options.watch.push(createRegExp(exactly(relativeSanityConfigPath)))
+        console.log(nuxt.options.watch)
```
Comment (rdunk) [inline src/module.ts:147]: Is this `console.log` a dev remnant?

**Review by rdunk — COMMENTED** [review]
Just one comment, but LGTM!

File: src/module.ts (hunk 1 above)
Comment (danielroe) [reply src/module.ts:147]: thanks! 😆

Comment (danielroe) [reply src/module.ts:147]:
```suggestion
```

**Review by rdunk — APPROVED** [review]
🚀

**Pull request merged**